File: 19_factorial.c
//C program to print factorial of a number

#include<stdio.h>

int main(){
    int num, fact;
    
    printf("Enter any integer: ");
    scanf("%d", &num);
    
    if (num<0){
        printf("Error! Factorial of negative numbers not possible.\n");
        return 1;
    }
    
    
    else if(num==0){
        fact==1;
    }
    
    else{
        for(int i=num; i>0; i--){
            fact*=1;
        }
        printf("\nFactorial of %d is %d\n", num, fact);
        return 0;
    }

File: 19_factorial.py
# Python program to calculate factorial of a number

#design user-defined functio nto calculate factorial
def calculate_factorial(number):
  if number==0 or number==1:
    return 1
  else:
    return number * calculate_factorial(number-1)

#main program starts here
#take input of number
num=int(input("Enter any integer: "))
#check if factorial valid and calculate factorial
if num<0:
  print("Factorial not valid for negative numbers!")
else:
  factorial = calculate_factorial(num)
  print(f"Factorial of entered number {num} is: {factorial}")
